subprojects/snapshots/src/main/java/org/gradle/internal/snapshot/impl/DirectorySnapshotter.java

```diff
@@ -214,17 +247,15 @@ private boolean isNotFileSystemLoopException(@Nullable IOException e) {
             return e != null && !(e instanceof FileSystemLoopException);
         }
 
-        private void addFileSnapshot(Path file, String name, BasicFileAttributes attrs) {
-            Preconditions.checkNotNull(attrs, "Unauthorized access to %", file);
-            HashCode hash = hasher.hash(file.toFile(), attrs.size(), attrs.lastModifiedTime().toMillis());
-            RegularFileSnapshot fileSnapshot = new RegularFileSnapshot(internedAbsolutePath(file), name, hash, FileMetadata.from(attrs));
-            builder.visit(fileSnapshot);
-        }
-
-        private String internedAbsolutePath(Path file) {
-            return stringInterner.intern(file.toString());
+        private String intern(String string) {
+            return stringInterner.intern(string);
         }
 
+        /**
+         * Returns whether we want to visit the given path during our walk, or ignore it completely,
+         * based on the directory/file excludes or the provided filtering predicate.
+         * Excludes won't mark this walk as `filtered`, only if the `predicate` rejects any entry.
+         **/
         private boolean isAllowed(Path path, String name, boolean isDirectory, @Nullable BasicFileAttributes attrs, Iterable<String> relativePath) {
```

Comment:
Reading the doc it seems to me that this method should be called `shouldVisit` or something, wdyt?